File: fintech_interview/factorial.c
#include <stdio.h>
/**
 * main - entry point of the program
 * description: finds the factorial of a iven number in C
 *
 * Return: always (0) sucess
 */
int main(void)
{
	int num;
	int i;
	unsigned long long factorial = 1;

	printf("enter a non negative integer: ");
	scanf("%d", &num);

	if (num < 0)
	{
		printf("factorial is not defined for negative numbers\n");
	}
	else if (num == 0)
	{
		printf("the factorial of 0 is 1\n");
	}
	else
	{
		for (i = 1; i <= num; i++)
		{
			factorial *= i;
		}
		printf("the factorial of %d is %llu", num, factorial);
	}
	return (0);
}

File: interview/factorial.py
# find the factorial of a given number in python
def factorial(n):
    result = 1
    for i in range(1, n + 1):
        result *= i
    return result

# Taking user input
num = int(input("Enter a number: "))

if num < 0:
    print("Factorial is not defined for negative numbers.")
else:
    print(f"The factorial of {num} is {factorial(num)}")
